### DISCUSSION TITLE
Update option picker 2 with values based on value in option picker 1

### DISCUSSION THREAD

**Discussion by renosyd-anders** [Budibase/budibase#10422] [Help] [answered]
Hi,

I have a Budibase DB with two tables: Containers and Pickups. Table Containers have fields like "Container type" and "Location". Examples:
Glass, London
Tires, London
Wood, London
Tires, Berlin
Metal, Berlin
Wood, Paris

Table Pickups is for new form entries and contains fields like "Container" and "Location".

In my new row side panel form, I have two option pickers with "Location" and "Containers". When picking a location, I would like the picker for container type to update according to the table Containers. Example: When choosing "Berlin" in picker 1, picker 2 should be updated with and only show "Tires" and "Metal".

![image](https://user-images.githubusercontent.com/116099444/234407041-5c2bb8ae-e609-475c-8e78-bc87cc9f1600.png)

Is that possible? I've read several docs, seen videos and searched discussions without luck.

**Comment by andz-bb** [comment, marked as answer]
hey @renosyd-anders 

This is possible by using a filtered data provider as the source for your options picker. Simply add a new data provider for the Container table inside your form and then set it up to filter by location where the location is equal to the value selected in your location form field. 

<img width="1470" alt="image" src="https://user-images.githubusercontent.com/110921612/234526156-c69689b4-9714-4bbd-8eb8-04d8dfbb4ae5.png">

Then when you have that data provider returning a filtered list of containers, change the Options source for the Container field to be the data provider. Select `Container Type` as the label and value for the picker and it will list the available container types for the selected location.

<img width="1466" alt="image" src="https://user-images.githubusercontent.com/110921612/234525376-4657be36-44a2-481e-b76f-061e464a3a8f.png">

For an extra bit of polish you can add an On Change action to the location picker so that it resets the container type value in case you had it set to something that wouldn't be available after the change.

<img width="1464" alt="image" src="https://user-images.githubusercontent.com/110921612/234531070-7692ab42-ed5d-4cef-a1da-16a6815b771a.png">

Here's the final result in action. Hope this helped!

![Screen Recording](https://user-images.githubusercontent.com/110921612/234531734-53baccc0-dec0-49ac-87e8-84e7677e6174.gif)

**Reply by renosyd-anders** [reply]
Hi @andz-bb, 

I'm so sorry for not answering before. I had to rebuild my app from scratch with your input due to too many changes. 

Thank you so much for your help. You really saved my a..!

**Reply by melohagan** [reply]
A small addition to this one as well is that if you are using relationships, you can now chain relationship pickers:

- [Filtering a relationship picker docs](https://docs.budibase.com/docs/filter-by-relationships#filter-the-list-of-options)
- [Cascading dropdown filters recipe](https://docs.budibase.com/docs/cascading-dropdown-filters)
